core/jvm/src/main/scala/zio/sql/expr.scala

```diff
@@ -210,6 +214,9 @@ trait ExprModule extends NewtypesModule with FeaturesModule with OpsModule {
 
   sealed case class FunctionDef[-A, +B](name: FunctionName) { self =>
 
+    def apply[Source, B1 >: B]()(implicit typeTag: TypeTag[B1]): Expr[Unit, Source, B1] =
```

Comment:
All the features that create a new `Expr` (rather than operating on an existing one) in some way introduce a new `Feature` type (that's the `F` parameter getting passed around everywhere). Can you please add a `Features.Function0` (like `Features.Source`) and use that instead of `Unit` for the `F` parameter?

As an `Expr[F, -A, +B]` models a function `A => B` (with some `Feature`s `F`) we can get rid of `Source` entirely and change the `A` (currently `Source`) to `Any` because a `Function0` can accept any value (it's ignoring everything in `Source`).

I think we should end up with something like 
def apply[B1 >: B]()(implicit typeTag: TypeTag[B1]): Expr[Features.Function0, Any, B1]

Reply:
Hello Rob, I'm having some issues applying this (all the others fixes have been applied). The next line (218) give me error. I attach a screenshot.
<img width="1015" alt="Screenshot 2020-11-27 at 18 18 57" src="https://user-images.githubusercontent.com/3865360/100472707-1600f800-30dd-11eb-9d57-09f2c5d62035.png">

Reply:
At the moment I commit without this change and it works.

Reply:
Ok, delete all the previous comments, I think that I did it! Look at my last commit.. Thanks!

Reply:
you can cast here instead of using the type ascription for self (e.g. `self.asInstanceOf[Func`... instead of `self:Func`...)

Reply:
look at my last comment ;)